### DISCUSSION TITLE
Сделать применение ключей формы при экспорте модели.

### DISCUSSION THREAD

**Discussion by SamArtGit** [PavelBlend/blender-xray#458] [Q&A] [answered]
При экспорте они сбрасываются.

![image](https://user-images.githubusercontent.com/56885075/150779785-1b71db3d-7852-4724-9968-a0d699843f89.png)

**Comment by PavelBlend** [comment]
@SamArtGit а как именно нужно сделать применение? То, что в 3д окне отображается, то и применять?
И в каких ситуациях может это пригодится?

**Comment by SamArtGit** [comment]
Вот сейчас делаю модель. Решил поправить воротник у капюшона чтобы при различных анимациях где наклоняется голова челюсть не проходила сквозь него. Перед экспортом мне надо создать Новую форму из смеси предыдущих и удалить по очереди все формы начиная с Базовой, чтобы не сбросились изменения. А так можно будет не морочиться и сразу экспортировать, по аналогии с разделёнными мешами в .object (аддон хоть и выдаёт предупреждение, но продолжает экспорт и объединяет модели) или триангуляцией (как-то раз забыл триангулировать модель, открываю Actor Editor и вижу, что аддон всё сделал за меня).

![image](https://user-images.githubusercontent.com/56885075/150798897-8ae5b89c-dadd-4a27-b0cd-660182773a6c.png)

![image](https://user-images.githubusercontent.com/56885075/150798990-cbc276ba-7779-4473-a70a-8cdf4cd79551.png)

**Comment by PavelBlend** [comment]
@SamArtGit хорошо, сделаю. Создал issue #459

**Comment by PavelBlend** [comment, marked as answer]
Сделал: https://github.com/PavelBlend/blender-xray/commit/c38bb793ff643285750bc48bd3353e6610ec4227

**Comment by shadept-GitHub** [comment]
Хотел спросить: а данную фичу для лицевой анимации возможно использовать? То есть при разных движениях рта просто изменять с  0 до 1 разные ползунки ключей формы.

**Reply by SamArtGit** [reply]
Да, но для Source и других современных движков. В Stalker такое не работает.

**Reply by shadept-GitHub** [reply]
Хм. То вот я и не совсем понимаю зачем данная фича есть кроме того чтобы исправлять косяки моделей с некоторыми позициями костей.

**Reply by PavelBlend** [reply]
@shadept-GitHub это не деструктивный способ редактирования. Можно создать несколько ключей и по очереди их экспортировать и смотреть в игре, какая модель лучше деформируется, и к оригинальной модели можно будет вернуться в любой момент.

И ещё без применения shape keys возникали разногласия: модель в блендере выглядит одним образом, а после экспорта в сдк уже по другому. Поэтому нужно применять shape keys.
